### PR TITLE
Add HM to cellGameDataCheck

### PR DESCRIPTION
**Pull request by clienthax**
Needed for Playstation Home to boot properly..

### CODE REVIEW

**Comment by Megamouse** [comment]
better also take a look at cellHddGameCheck

File: rpcs3/Emu/Cell/Modules/cellGame.cpp [hunk 1]
```diff
@@ -594,7 +594,7 @@ error_code cellGameDataCheck(u32 type, vm::cptr<char> dirName, vm::ptr<CellGameC
 	{
 		switch (type)
 		{
-		case CELL_GAME_GAMETYPE_HDD: return cat != "HG"sv && cat != "AM"sv && cat != "AP"sv && cat != "AT"sv && cat != "AV"sv && cat != "BV"sv && cat != "WT"sv;
+		case CELL_GAME_GAMETYPE_HDD: return cat != "HG"sv && cat != "HM"sv && cat != "AM"sv && cat != "AP"sv && cat != "AT"sv && cat != "AV"sv && cat != "BV"sv && cat != "WT"sv;
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellGame.cpp:597]: Why complicate it even further? simply != DG != GD && !.empty()

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellGame.cpp:597]: Because there could be garbage in the string obviously

Comment (elad335) [reply rpcs3/Emu/Cell/Modules/cellGame.cpp:597]: May add such "garbage" checks in SFO loader itself if we are so anxious about it.

**Comment by Megamouse** [comment]
superseded by #11745

**Pull request closed without merging**